Source/SwiftLintFramework/Rules/WeakDelegateRule.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  WeakDelegate.swift
+//  SwiftLint
+//
+//  Created by Olivier Halligon on 11/08/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct WeakDelegateRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "weak_delegate",
+        name: "Weak Delegate",
+        description: "Delegates should be weak to avoid reference cycles.",
+        nonTriggeringExamples: [
+            "class Foo {\n  weak var delegate: SomeProtocol?\n}\n",
+            "class Foo {\n  weak var someDelegate: SomeDelegateProtocol?\n}\n",
+            "class Foo {\n  weak var delegateScroll: ScrollDelegate?\n}\n",
+            // We only consider properties to be a delegate if it has "delegate" in its name
+            "class Foo {\n  var scrollHandler: ScrollDelegate?\n}\n",
+            // Only trigger on instance variables, not local variables
+            "func foo() {\n  var delegate: SomeDelegate\n}\n"
+        ],
+        triggeringExamples: [
+            "class Foo {\n  var delegate: SomeProtocol?\n}\n",
+            "class Foo {\n  var scrollDelegate: ScrollDelegate?\n}\n",
+            "class Foo {\n  var delegateScroll: ScrollDelegate?\n}\n",
+        ]
+    )
+
+    public func validateFile(file: File,
+                             kind: SwiftDeclarationKind,
+                             dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard kind == .VarInstance else {
+            return []
+        }
+
+        // Check if name contains "delegate"
+        let name = (dictionary["key.name"] as? String) ?? ""
+        let isDelegate = name.lowercaseString.rangeOfString("delegate") != nil
+        guard isDelegate else { return [] }
+
+        // Check if non-weak
+        let attributes = (dictionary["key.attributes"] as? [SourceKitRepresentable])?
```

Comment:
Do you have a way to check if the declaration is Optional and a Protocol from source kitten?  Perhaps you can simplify it by using a regex 

```swift
 public func validateFile(file: File) -> [StyleViolation] {
        let lines = file.lines
        return file.matchPattern(".*[dD]elegate.*:*\\s*\\?").flatMap { range, syntaxKinds in
            let location = Location(file: file, characterOffset: range.location)
            guard let lineIndex = location.line,
                   let currentLine = lines.filter({ $0.index == lineIndex }).first
                   else { return nil }
            guard !currentLine.content.containsString("weak") else { return nil }

            return StyleViolation(ruleDescription: self.dynamicType.description,
                severity: configuration.severity,
                location: location)
        }
    }

```

![image](https://cloud.githubusercontent.com/assets/6539412/17614361/989bfadc-601a-11e6-8876-d98d7179fd81.png)